match pipeline visualization graph with latest UX

karthikjeeyar opened the pull request:
Contains the new layout for pipeline and pipeline run visualization

**Pipeline Visualization:**

![screenshot-localhost-9000-2019 09 09-23-42-00](https://user-images.githubusercontent.com/9964343/64557341-fe661400-d35e-11e9-8fb7-efaacef17dae.png)

**Pipelinerun Visualization**
![screenshot-localhost-9000-2019 09 09-23-43-01 (1)](https://user-images.githubusercontent.com/9964343/64557350-02923180-d35f-11e9-9412-06f31b318f89.png)

Fixes: ODC-1460

christianvogt commented:
With the following status:
![image](https://user-images.githubusercontent.com/14068621/64567647-b3c8a580-d326-11e9-9416-3738f3bd628d.png)

I'd expect the last two tasks to show a grey state.
![image](https://user-images.githubusercontent.com/14068621/64567632-ad3a2e00-d326-11e9-93f0-475276eeeea4.png)

christianvogt commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.tsx:
```diff
@@ -81,22 +84,16 @@ export const PipelineVisualizationTask: React.FC<PipelineVisualizationTaskProp>
-      }}
-    >
+    <li className={cx('odc-pipeline-vis-task')}>
+      <div className="connector" />
```
```suggestion
      <div className="odc-pipeline-vis-task__connector" />
```

christianvogt commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.scss:
```diff
@@ -2,23 +2,51 @@ $border-color: var(--pf-global--BorderColor--light-100);
-    position: relative;
-    top: 1px;
+  }
+  &> .connector {
```
```suggestion
  &__connector {
```

karthikjeeyar replied:
updated it

christianvogt commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.scss:
```diff
@@ -46,15 +79,15 @@ $gutter: 1.8em;
-    margin: 0 4px;
+    margin-left: 7px;
     width: 1.5em;
+    height:1.5em;
```
```suggestion
    height: 1.5em;
```

karthikjeeyar replied:
updated

karthikjeeyar commented:
> Zooming in, the lines look funky. But even at 100% zoom, some lines look fuzzy.

![screenshot-localhost-9000-2019 09 10-16-31-59](https://user-images.githubusercontent.com/9964343/64608800-0ff3fe00-d3e9-11e9-9fd7-4789d3c3bf38.png)

updated the fuzzy lines

abhinandan13jan commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.tsx:
```diff
@@ -81,22 +84,16 @@ export const PipelineVisualizationTask: React.FC<PipelineVisualizationTaskProp>
         name={task.name || ''}
         namespace={namespace}
         status={task.status || { duration: '', reason: 'pending' }}
+        isPipelineRun={!!task.status}
```
You need to update this area `task.status` will be non-existent for both cases, tasks of a `pipeline` and `pending/cancelled` tasks of  a `pipelinerun`. So, it will not be a distinguishing factor of a `pipeline` vs `run` for all cases. It would be better to pass the `pipelineRun` as an optional prop to the `PipelineVisualizationGraph`. Using this `prop` you should be able to distinguish between `Pipeline` and `PipelineRun` Tasks as well as `Pending` and `Cancelled` task by putting it into the `Pipelinerunfilterreducer` functionality.

karthikjeeyar replied:
now using the pipelineRunStatus to distinguish between pipeline and pipelineRun

karthikjeeyar commented:
![screenshot-localhost-9000-2019 09 10-19-57-05](https://user-images.githubusercontent.com/9964343/64625128-480b3900-d409-11e9-8f60-c5a161c4f6d4.png)
![screenshot-localhost-9000-2019 09 10-19-57-21](https://user-images.githubusercontent.com/9964343/64625131-493c6600-d409-11e9-9934-b04ced0f641b.png)
Fixed the cancelled state issue
cc: @christianvogt

karthikjeeyar commented:
/retest

serenamarie125 commented:
@karthikjeeyar what are you using for icons for the specific task states ? I think we need to make a few updates!

serenamarie125 commented:
@karthikjeeyar @christianvogt Here are the associated icons.  If you need help with the names or if they are PF/FA, let me know!

![Pipeline_Palette_4 2Final (2)](https://user-images.githubusercontent.com/12504403/64628218-fe95fc00-d3be-11e9-8ca9-9d6e9bc04321.jpg)

karthikjeeyar commented:
Thanks @serenamarie125, i will update it

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.scss:
```diff
@@ -69,32 +80,45 @@ $gutter: 1.8em;
     &.is-text-center {
       text-align: center;
+      max-width: 100px;
+      margin: 0 var(--pf-global--spacer--md);
```
With the pills being `120px` wide and the `--pf-global-spacer--md` being `16px`, the `max-width` is never reached (120 - 16 * 2 = 88) unless the variable is changed.

Are we just looking to avoid touching the sides? I think we can achieve this with:
```suggestion
      margin: 0 auto;
```
While keeping the `max-width` to prevent it from squishing against the sides.

Is there a style guideline that says we should be using this variable to space things?

andrewballantyne replied:
Also, @serenamarie125 the Visual Spec document on the ticket states the text should be 75px wide - is that only when an icon is in play?

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.scss:
```diff
@@ -46,15 +57,15 @@ $gutter: 1.8em;
 
   &__status {
-    margin: 0 4px;
+    margin-left: 7px;
```
I think we should throw all the margin in this basket as it's spacing purely for the icon...
```suggestion
    margin: 0 3px 0 7px;
```

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.scss:
```diff
@@ -1,43 +1,54 @@
-    top: 1px;
+    flex-direction: row;
+    align-items: center;
+    justify-content: center;
```
I think we want to left align the content
```suggestion

```

Justify Center:
<img width="153" alt="Screen Shot 2019-09-10 at 1 20 29 PM" src="https://user-images.githubusercontent.com/8126518/64635495-cfd35200-d3cd-11e9-90b0-0a1fc5d12f96.png">

Left-Aligned (omit style):
<img width="163" alt="Screen Shot 2019-09-10 at 1 20 36 PM" src="https://user-images.githubusercontent.com/8126518/64635522-deba0480-d3cd-11e9-9316-ac49302bfbe7.png">

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.scss:
```diff
@@ -46,15 +57,15 @@ $gutter: 1.8em;
-    font-size: 1.1em;
   }
 
   &__title {
```
The title is also exceeding the `75px` width set in the design spec.

![image](https://user-images.githubusercontent.com/8126518/64636192-2db46980-d3cf-11e9-936e-367e87ce785a.png)

Recommend swapping the `flex-grow` and `flex-shrink` for:
`flex: 0 0 75px;`

karthikjeeyar commented:
![screenshot-localhost-9000-2019 09 10-23-06-51](https://user-images.githubusercontent.com/9964343/64638079-ba3c4780-d422-11e9-925a-6065519f3ede.png)
Updated the icons and added the changes suggested by @andrewballantyne 
cc: @serenamarie125

andrewballantyne commented on frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.scss:
```diff
@@ -46,55 +56,66 @@ $gutter: 1.8em;
     &:hover {
       cursor: pointer;
     }
     &.is-text-center {
```
Oops, I may have forgot to add some extra styles I had for this. I think we just visually broke the pills on the pipeline overview screen.

```
      flex-basis: 0; // get rid of the 75px width restriction
      flex-grow: 1; // fill the space it needs (up to a 100px)
```

andrewballantyne commented:
/lgtm

andrewballantyne commented:
I'll work to add the status bars and fixup the tooltip (which I'll create a ticket for)

andrewballantyne commented:
/retest

andrewballantyne commented:
/retest

andrewballantyne commented:
/test e2e-aws-console-olm

christianvogt commented:
/approve

vikram-raj commented:
/test e2e-aws-console-olm
